docs/DeveloperGuide.md

```diff
@@ -213,25 +319,38 @@ The `Storage` component,
 
 ### Common classes
 
-Classes used by multiple components are in the `seedu.addressbook.commons` package.
+Classes used by multiple components are in the [`seedu.addressbook.commons`](https://github.com/AY2223S1-CS2103T-W17-2/tp/tree/master/src/main/java/seedu/pennywise/commons) package.
+
+<p align="right">
+    <a href="#top">Back to Top </a>
+</p>
 
 --------------------------------------------------------------------------------------------------------------------
 
 ## **Implementation**
 
-This section describes some noteworthy details on how certain features are implemented.
+This section describes some noteworthy details on how certain features are implemented, as well as the considerations that goes behind the implementations.
+You can also find some of our proposed implementations for challenging features that we will focus on in later iterations.
 
 ### Summarise Entries
 
-The `summary` command is implemented by the `SummaryCommandParser` and `SummaryCommand` classes
+The `summary` feature provides 3 simple statistic, including the total expenditure, total income and total balance,
+helping students to have a quick overview of their expenditure and income. If the command is optionally provided with a
+year and month field, the summary statistics will be contained to only the entries occurring in the specified year and month.
+
+#### How the `summary` command works
 
-`SummaryCommandParser` class is responsible for parsing the parameter received from the user
+The `summary` command is implemented by the `SummaryCommandParser` and `SummaryCommand` classes.
 
-`SummaryCommand` class is responsible for generating the summary statistic for the specified duration
+`SummaryCommandParser` class is responsible for parsing the parameter received from the user.
 
-Below is a sequence diagram and explanation of how the SummaryCommand is executed.
+`SummaryCommand` class is responsible for generating the summary statistic for the specified duration.
+
+Below is a sequence diagram and explanation of how the `summary` command is executed.
 ![Interactions Inside the Logic Component for the `summary mo/2022-08` Command](images/SummarySequenceDiagram.png)
 
+<div markdown="span" class="alert alert-info">:information_source: **Note:** The lifeline for `SummaryCommandParser` should end at the destroy marker (X) but due to a limitation of PlantUML, the lifeline reaches the end of diagram.</div>
+
 Step 1.The user enters `summary mo/2022-08` command in the main window
```

Comment:
Should we delete all these `Step x.` lines for all implementations? @AY2223S1-CS2103T-W17-2/developers

Reply:
Maybe change to a bullet list?

Reply:
Hmm yea I think that would make it neater but it might still be considered repetitive to readers

Reply:
Think numbered lists should be okay, since our dev guide's purpose is just to comprehensively allow the readers to understand how the functions/commands work